packages/amplify_api_rest/android/src/main/kotlin/com/amazonaws/amplify/amplify_api_rest/AmplifyApiRestPlugin.kt

```diff
@@ -0,0 +1,311 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazonaws.amplify.amplify_api_rest
+
+import android.os.Handler
+import android.os.Looper
+import androidx.annotation.NonNull
+import androidx.annotation.VisibleForTesting
+import com.amazonaws.amplify.amplify_api_rest.types.FlutterAuthFailureMessage
+import com.amazonaws.amplify.amplify_api_rest.types.FlutterRestInputs
+import com.amazonaws.amplify.amplify_api_rest.types.FlutterRestResponse
+import com.amplifyframework.AmplifyException
+import com.amplifyframework.api.ApiException
+import com.amplifyframework.api.aws.AWSApiPlugin
+import com.amplifyframework.api.rest.RestOperation
+import com.amplifyframework.api.rest.RestOptions
+import com.amplifyframework.api.rest.RestResponse
+import com.amplifyframework.core.Amplify
+import com.amplifyframework.core.Consumer
+
+import io.flutter.embedding.engine.plugins.FlutterPlugin
+import io.flutter.plugin.common.MethodCall
+import io.flutter.plugin.common.MethodChannel
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler
+import io.flutter.plugin.common.MethodChannel.Result
+import io.flutter.plugin.common.PluginRegistry.Registrar
+import kotlin.reflect.KFunction3
+import kotlin.reflect.KFunction4
+
+class AmplifyApiRestPlugin: FlutterPlugin, MethodCallHandler {
+
+  private lateinit var channel : MethodChannel
+  private var operationsMap : HashMap<String, RestOperation> = HashMap()
+
+  override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
+    channel = MethodChannel(flutterPluginBinding.getFlutterEngine().getDartExecutor(), "com.amazonaws.amplify/amplify_api_rest")
+    channel.setMethodCallHandler(this)
+    Amplify.addPlugin(AWSApiPlugin())
+  }
+
+  companion object {
+    val LOG = Amplify.Logging.forNamespace("amplify:flutter:api_rest")
+
+    @JvmStatic
+    fun registerWith(registrar: Registrar) {
+      val channel = MethodChannel(registrar.messenger(), "com.amazonaws.amplify/amplify_api_rest")
+      channel.setMethodCallHandler(AmplifyApiRestPlugin())
+      Amplify.addPlugin(AWSApiPlugin())
+      LOG.info("Added AWSApiPlugin")
+    }
+
+  }
+
+  override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
+    channel.setMethodCallHandler(null)
+  }
+
+  @VisibleForTesting
+  override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
+    
+    if(call.method == "cancel"){
```

Comment:
Make sure to use whitespace after keywords and before curly braces, in all places:
```suggestion
    if (call.method == "cancel") {
```

Reply:
Hey @jamesonwilliams is there a way to ensure that those syntax rules are automatically enforced?

Reply:
Such as being auto applied whenever I save the file?

Reply:
Try `Ctrl + Command + Shift + L`. It will apply the default reformatting rules in your AS. We need to align on a team wide linter rules.

Reply:
Ah thanks @Amplifiyer .  I applied that and it didn't change the spacing to what Jameson recommended.  I recommend we come back to this later when we've aligned on a team wide linter rule set and can just apply that to all our files in one go. @jamesonwilliams

Reply:
Do you use VS Code for Kotlin, or Android Studio, @fjnoyp? Regardless, I'll plan to onboard `ktlint` so we can catch these types of things before code review. Checkout the formatting options in the preferences under Android Studio. You can configure it to automatically format on save. VS Code does that out of the box.

Kotlin code style is more cut and dry than some other languages. It has its own conventions document: https://kotlinlang.org/docs/reference/coding-conventions.html#coding-conventions.